docs/input.rst

```diff
@@ -57,39 +57,33 @@ a future ``read()`` or ``take()``::
 Accessing the data samples
 ~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-After calling the read/take operations there are several ways to access the
-data samples. The simplest uses :meth:`Input.valid_data_iterator` to iterate
-through all the samples with valid data and calls :meth:`SampleIterator.get_dictionary()`
-to retrieve all the fields in the sample:
+After calling the read/take operations, :attr:`Input.samples` contains the data
+samples:
 
 .. testcode::
 
-   for sample in input.valid_data_iterator:
-      print(sample.get_dictionary())
+   for sample in input.samples:
+      if sample.valid_data:
+         print(sample.get_dictionary())
 
-Use :meth:`Input.data_iterator` to also access samples that contain
-meta-data only (see next section, :ref:`Accessing the SampleInfo`):
+:meth:`SampleIterator.get_dictionary()` retrieves all the fields of a sample.
```

Comment:
Given that the SampleIterator class is no longer exposed to the user (they have input._samples, which internally may provide a SampleIterator) maybe we shouldn't include it in the documentation

Reply:
SampleIterator contains all the methods for data access get_number, get_dictionary, etc. We can't hide it.